vault/resource_aws_auth_backend_sts_role_test.go

```diff
@@ -117,6 +165,11 @@ func testAccAWSAuthBackendSTSRoleCheck_attrs(backend, accountID, stsRole string)
 		attrs := map[string]string{
 			"sts_role": "sts_role",
 		}
+		// Only check external_id if Vault version is >= 1.17.0
+		if provider.IsAPISupported(testProvider.Meta(), provider.VaultVersion117) {
+			attrs[consts.FieldExternalID] = consts.FieldExternalID
+		}
```

Comment:
@vinay-gopalan Since I added this version check in `testAccAWSAuthBackendSTSRoleCheck_attrs`, I didn't end up using something like the [`SkipFunc`](https://github.com/hashicorp/terraform-provider-vault/blob/63c5a309b78816c36f40d2c028b139e1361ba92a/vault/resource_gcp_secret_backend_test.go#L51-L63) example you sent.

I could add `SkipFunc` to the applicable `external_id` test steps (or switch to using `SkipFunc` instead of this logic in `testAccAWSAuthBackendSTSRoleCheck_attrs`), but wasn't sure which approach was best / most consistent with the rest of the repo.

Reply:
We synced and decided not to modify this `testAccAWSAuthBackendSTSRoleCheck_attrs`, and instead create a separate test for `withExternalID` that only runs on Vault versions >= 1.17 and uses the `TestCheckResourceAttr` helper pattern. See https://github.com/hashicorp/terraform-provider-vault/pull/2370/commits/67bb7aa2f0b849048741f005c85d535181bfcbf6 for changes